### PR TITLE
Name workflows and create codeowners for permissions

### LINKED ISSUE #142: Permissions and Responsibilities Structure
GitHub offers the following permission levels for organization repositories ([docs](https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-roles-for-an-organization#repository-roles-for-organizations)):

> * Read: Recommended for non-code contributors who want to view or discuss your project
> * Triage: Recommended for contributors who need to proactively manage issues and pull requests without write access
> * Write: Recommended for contributors who actively push to your project
> * Maintain: Recommended for project managers who need to manage the repository without access to sensitive or destructive actions
> * Admin: Recommended for people who need full access to the project, including sensitive and destructive actions like managing security or deleting a repository

I propose the following structure of roles for Vivarium and Vivarium Core (I'm ignoring the other Vivarium repos for now):

* Maintainers: These are the core "owners" of the Vivarium project. They have the `owner` role in the Vivarium Collective organization and the `admin` role in Vivarium Core. They are responsible for enforcing the code of conduct, responding to security vulnerabilities, and adjudicating disputes.
* Collaborators: These are trusted contributors who help develop Vivarium projects. They have the `member` role in the Vivarium Collective organization and the `write` role in Vivarium Core. They are empowered to manage issues, open PRs, and otherwise develop Vivarium Core.
* Public: Anyone can read the Vivarium Core repo, and anyone can contribute by forking the repository and opening a PR. Anyone can also open or comment on a discussion thread or issue.

Remaining Questions:

* Should we require PRs to be approved by certain people? This would probably mean maintainers approve PRs, but collaborators can merge them.
* The Apache CLA is based around the idea of an entity to whom contributors grant a license to use their contributions. Who should this entity be? Maybe the maintainers?

### PR DESCRIPTION
**Pull request by U8NWXD**
Fixes #142 by:

* Naming GitHub Actions workflows so we can require that certain workflows pass before PRs can be merged.
* Creating a CODEOWNERS file so that maintainers have to approve all PRs. See [GitHub's docs](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#codeowners-syntax) for syntax

-----------------------------------------------------------------------

By creating this pull request, I agree to the Contributor License
Agreement, which is available in `CLA.md` at the top level of this
repository.

### CODE REVIEW

**Review by eagmon — APPROVED** [review]

**Comment by eagmon** [comment]
@U8NWXD -- Can you allow the maintainers to merge changes without approval?

**Comment by U8NWXD** [comment]
done!

**Comment by U8NWXD** [comment]
Screen Shot 2022-01-27 at 15 19 24

As a maintainer you should see something like this with a red "Merge pull request" button that lets you bypass the normal checks

**Pull request merged**
